components/common/src/test/java/com/hotels/styx/api/EarlyReturnTest.java

```diff
@@ -0,0 +1,99 @@
+/*
+  Copyright (C) 2013-2019 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.api;
+
+import org.testng.annotations.Test;
+import reactor.core.publisher.Mono;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static com.hotels.styx.api.HttpResponseStatus.OK;
+import static com.hotels.styx.api.LiveHttpRequest.post;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+public class EarlyReturnTest {
+    @Test
+    public void bodyIsConsumedWhenReturningEarlyWithError() {
```

Comment:
Do we have a test that shows the opposite behaviour?  Body is not consumed when not returning early.

Reply:
Not sure what that would be? These tests test what happens when convenience methods are called. The opposite would be to test that nothing happens when you don't call any methods. 

Perhaps the test names are unclear? Please let me know.